### ISSUE TITLE
createElement setting?

### ISSUE THREAD

**Issue by ljharb** [yannickcr/eslint-plugin-react#1074]
We allow `createClass` to be overridden with a [setting](https://github.com/yannickcr/eslint-plugin-react#configuration) - should we allow `createElement` to be configured as well?

If so, any rules referencing `React.createElement` should be sure to use `<pragma>.<createElement>`

**Comment by TrySound** [comment]
This works for me
```yml
settings:
  react:
    pragma: 'createElement'

rules:
  no-unused-vars: 'error'
  react/jsx-uses-react: 'error'
  react/jsx-uses-vars: 'error'
```

**Comment by ljharb** [comment]
Doesn't that change `React` to `createElement`, which isn't the same thing? Or is the default pragma `React.createElement`?

**Comment by TrySound** [comment]
I guess pragma here is what imported. Maybe this may conflicts with createClass. But it's still deprecated and doesn't matter IMHO. So this way just works.

**Comment by ljharb** [comment]
It's important to note the caveat that it only works when not using anything off of `React` except for `createElement` - don't forget `React.Component`, `React.Children`, `React.isValidElement`, `React.PropTypes`, etc.

**Comment by TrySound** [comment]
Maybe. In my opinion these cases should be processed with flow or typescript. It's just an api, not syntax stuff.

**Comment by ljharb** [comment]
Neither of those things is a part of the language, and "types" is entirely unrelated to the reasons eslint-plugin-react needs to know about the `React` and `createClass` pragmas (which relate to using alternative renderers to React itself).

**Comment by jkrems** [comment]
Would it make sense to also allow `h(...)` as an alternative to `React.createElement(...)`? It's a relatively common pattern exposed by preact.

**Comment by ljharb** [comment]
It seems like we’d need a setting to specify “createElement” *and* whether it should chain it off the pragma or not.
